apps/web/src/components/CheckoutForm.tsx

```diff
@@ -0,0 +1,201 @@
+        fontFamily: "Source Code Pro, monospace",
+        "::placeholder": {
+          color: "#aab7c4"
+        },
+        padding
+      },
+      invalid: {
+        color: "#9e2146"
+      }
+    }
+  };
+};
+
+const FormContainer: AnyStyledComponent = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-content: center;
+  justify-content: center;
+  align-items: center;
+
+  margin: 3rem 1rem 0 1rem;
+`;
+
+const ElementContainer: AnyStyledComponent = styled.div`
+  width: 100%;
+  input,
+  .StripeElement {
+    display: block;
+    margin: 10px 0 20px 0;
+    max-width: 500px;
+    padding: 8px 12px;
+    font-size: 1em;
+    font-family: "Source Code Pro", monospace;
+    box-shadow: rgba(50, 50, 93, 0.14902) 0px 1px 3px,
+      rgba(0, 0, 0, 0.0196078) 0px 1px 0px;
+    border: 0;
+    outline: 0;
+    border-radius: 0.5rem;
+    background: white;
+  }
+`;
+
+const ErrorContainer: AnyStyledComponent = styled.div`
+  display: flex;
+  align-items: center;
+  align-content: center;
+
+  color: #f56565;
+`;
+
+type CheckoutProps = ReactStripeElements.InjectedStripeProps & {
+  tag: MembershipTypes;
+};
+
+type setString = React.Dispatch<React.SetStateAction<string>>;
+type setBoolean = React.Dispatch<React.SetStateAction<boolean>>;
+
+const CheckoutFormBase: React.FC<CheckoutProps> = (
+  props: CheckoutProps
+): JSX.Element => {
+  const [cardElement, setCardElement] = useState<any>(undefined);
+  const [clientSecret, setClientSecret]: [string, setString] = useState<string>(
+    ""
+  );
+  const [error, setError]: [string, setString] = useState<string>("");
+  const [loading, setLoading]: [boolean, setBoolean] = useState<boolean>(false);
+
+  const [getMembership] = useGetMembershipMutation();
+
+  const handleSubmit: (ev: React.FormEvent) => void = async (
+    ev: React.FormEvent
+  ): Promise<void> => {
+    // We don't want to let default form submission happen here, which would refresh the page.
+    ev.preventDefault();
+    setError("");
+    setLoading(true);
+    if (!props.stripe) {
+      setError("Stripe.js hasn't loaded yet.");
+      return;
+    }
+    let secret: string = clientSecret;
+
+    if (!secret) {
+      const result: ExecutionResult<
+        GetMembershipMutation
+      > = await getMembership({
+        variables: { membership: props.tag }
+      });
+      const data = result.data;
+
+      if (!data) {
+        if (result.errors) {
+          setError(result.errors[0].message || "Unknown Error occurred.");
+        }
+
+        return;
+      }
+      secret = data.startMembershipTransaction.clientSecret;
+      setClientSecret(secret);
+    }
+
+    const response: stripe.PaymentIntentResponse = await props.stripe.handleCardPayment(
+      secret,
+      cardElement
+    );
+
+    if (response.error) {
+      setError(response.error.message!);
+    }
+    setLoading(false);
+    console.log(response);
+  };
+
+  if (error && loading) {
+    setLoading(false);
```

Comment:
This shouldn't need to be checked for

Reply:
This is a safety check in case there is an error that is set, but loading is not reset so they the button becomes disabled. I can try to find a smarter way to do the check if thats what you're looking for